src/main/java/org/opensearch/security/support/HostAndCidrMatcher.java

```diff
@@ -66,7 +63,7 @@ public boolean matchesCidr(InetAddress address) {
             return cidrMatchers.stream().anyMatch(cidrAddress -> cidrAddress.contains(addressString));
         } catch (Exception e) {
             log.warn("Failed to process IP address {}: {}", address, e.getMessage());
-            return false;
+            throw new RuntimeException("Invalid Address format used");
```

Comment:
I'd recommend to throw the original exception e as cause. Alternatively, one could also remove the try block alltogether.

Reply:
ack, removed the block for simplicity.